crates/sparrow-runtime/src/prepare/column_behavior.rs

```diff
@@ -311,6 +376,52 @@ impl ColumnBehavior {
     }
 }
 
+/// Converts to the expected time unit using the time multiplier.
+fn scale_to_time_multiplier(
+    time: &ArrayRef,
+    time_multiplier: Option<i64>,
+) -> error_stack::Result<ArrayRef, Error> {
+    if let Some(time_multiplier) = time_multiplier {
+        let error = || Error::ConvertTime(time.data_type().clone());
+        match time.data_type() {
+            DataType::UInt8
+            | DataType::UInt16
+            | DataType::UInt32
+            | DataType::UInt64
+            | DataType::Int8
+            | DataType::Int16
+            | DataType::Int32
+            | DataType::Int64 => {
+                let time = arrow::compute::cast(&time, &DataType::Int64)
+                    .into_report()
+                    .change_context_lazy(error)?;
+                Ok(
+                    arrow::compute::multiply_scalar_dyn::<Int64Type>(
+                        time.as_ref(),
+                        time_multiplier,
+                    )
+                    .into_report()
+                    .change_context_lazy(error)?,
+                )
+            }
+            DataType::Float16 | DataType::Float32 | DataType::Float64 => {
+                let time = arrow::compute::cast(&time, &DataType::Float64)
+                    .into_report()
+                    .change_context_lazy(error)?;
+                Ok(arrow::compute::multiply_scalar_dyn::<Float64Type>(
+                    time.as_ref(),
+                    time_multiplier as f64,
+                )
+                .into_report()
+                .change_context_lazy(error)?)
+            }
+            _ => Ok(time.clone()),
+        }
+    } else {
+        Ok(time.clone())
+    }
+}
```

Comment:
A lot of this is halfway duplicated from the `prepare_batch` code. It's not a complete  duplicate, but there's definitely room to simplify, but it would require more refactoring of the `column_behavior` code. Not sure how much we want to do now.

Reply:
The best options seem to be either:

1. Drop the column behavior stuff and have both paths call `prepare_batch` (the newer code).
2. Add a column behavior corresponding to `numeric_to_timestamp` which `prepare_batch` uses for the timestamp case.

Either would let us avoid adding as much code to `column_behavior`.